### PR TITLE
chore: comments analytics

### PR DESCRIPTION
**Pull request by rephus**
Closes: #9004

### Description:

Screenshot from 2024-02-27 15-39-32
Screenshot from 2024-02-27 15-39-58

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/CommentService/CommentService.ts [hunk 1]
```diff
@@ -145,6 +155,13 @@ export class CommentService {
                 "You don't have access to the space this dashboard belongs to",
             );
         }
+        analytics.track({
+            event: 'comment.resolved',
+            userId: user.userUuid,
+            properties: {
+                // TODO if you want to add properties, we need a getCommentById method (we can refactor getCommentOwner)
```
Comment (joaoviana) [inline packages/backend/src/services/CommentService/CommentService.ts:162]: good point, like a 
`isOwner` or something?

File: packages/backend/src/services/CommentService/CommentService.ts [hunk 2]
```diff
@@ -184,5 +201,13 @@ export class CommentService {
 
             throw new ForbiddenError();
         }
+
+        analytics.track({
+            event: 'comment.deleted',
+            userId: user.userUuid,
+            properties: {
+                // TODO if you want to add properties, we need a getCommentById method (we can refactor getCommentOwner)
```
Comment (joaoviana) [inline packages/backend/src/services/CommentService/CommentService.ts:209]: `isOwner`?

File: packages/backend/src/services/CommentService/CommentService.ts (hunk 1 above)
Comment (rephus) [reply packages/backend/src/services/CommentService/CommentService.ts:162]: I was thinking refactor the `getCommentOwner` to return the whole comment object, to make it more generic,  and check userUuid in the service itself

**Comment by rephus** [comment]
I  think we can merge this as it si and revisit once we have mentions and nofications in place

File: packages/backend/src/models/CommentModel/CommentModel.ts [hunk 3]
```diff
@@ -19,6 +22,79 @@ export class CommentModel {
...
+
+            const structuredComment: Comment = {
+                commentId: comment.comment_id,
+                text: comment.text,
+                replyTo: comment.reply_to ?? undefined,
+                user: { name: `${comment.first_name} ${comment.last_name}` },
+                createdAt: comment.created_at,
+                resolved: comment.resolved,
+                replies: [],
+                canRemove:
```
Comment (joaoviana) [inline packages/backend/src/models/CommentModel/CommentModel.ts:49]: I think we just need to re-add the `textHtml` here

**Review by joaoviana — COMMENTED** [review]
Left one comment. 
The `hasMention` can be done after

I will add another issue for analytics events for notifications

EDIT: here it is the issue for notifications: https://github.com/lightdash/lightdash/issues/9121

File: packages/backend/src/models/CommentModel/CommentModel.ts (hunk 3 above)
Comment (rephus) [reply packages/backend/src/models/CommentModel/CommentModel.ts:49]: that is on a separate branch... sorry I'm not doing the things in the right order...

**Comment by rephus** [comment]
ok, let's pause this until everything else is merged

**Comment by joaoviana** [comment]
> ok, let's pause this until everything else is merged

sounds good @rephus

File: packages/backend/src/models/CommentModel/CommentModel.ts (hunk 3 above)
Comment (rephus) [reply packages/backend/src/models/CommentModel/CommentModel.ts:50]: updated with html

File: packages/backend/src/models/CommentModel/CommentModel.ts [hunk 4]
```diff
@@ -19,6 +22,81 @@ export class CommentModel {
         this.database = deps.database;
     }
 
+    private static parseComments(
+        commentsWithUsers: (DbDashboardTileComments &
+            Pick<DbUser, 'first_name' | 'last_name'>)[],
```
Comment (joaoviana) [inline packages/backend/src/models/CommentModel/CommentModel.ts:27]: 👍

**Pull request merged**